src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -182,90 +188,47 @@ protected virtual void ProcessTypes(AssemblyDefinition assembly, XPathNavigator
                         continue;
                 }
 
-                TypeDefinition type = assembly.MainModule.GetType(fullname);
-
-                if (type == null && assembly.MainModule.HasExportedTypes)
```

Comment:
Could you add a TODO that we might need a `ProcessExportedType` equivalent (we will need to mark the type forwarder if this went through a type forwarder).